### PR TITLE
GUACAMOLE-107 Remove references to old website and Apachify project name

### PR DESCRIPTION
**Pull request by jmuehlner**
It looks like only the README had to be touched here.

### CODE REVIEW

File: README [hunk 1]
```diff
@@ -4,15 +4,11 @@
 ------------------------------------------------------------
 
 This README is intended to provide quick and to-the-point documentation for
-technical users intending to compile parts of Guacamole themselves.
+technical users intending to compile parts of Apache Guacamole themselves.
 
-Source archives are available from the files section of the main project page:
```
Comment (mike-jumper) [inline README]: This should probably still be here, but instead refer to the "downloads" section. It is still valid, though definitely should not point to SourceForge.

Comment (jmuehlner) [inline README]: Ok

File: README [hunk 2]
```diff
@@ -6,11 +6,16 @@
...
+Source archives are available from the downloads section of the project website:
+ 
+    http://guacamole.incubator.apache.org/
+
+A full manual is available as well:
 
     http://guacamole.incubator.apache.org/doc/gug/
 
 
+
```
Comment (mike-jumper) [inline README]: Why did you add this blank line?

Comment (jmuehlner) [inline README]: :face_with_head_bandage:
